src/shared/condition.c

```diff
@@ -177,6 +178,27 @@ static int condition_test_user(Condition *c) {
         return id == getuid() || id == geteuid();
 }
 
+static int condition_test_control_group_controller(Condition *c) {
+        int r;
+        CGroupMask system_mask, wanted_mask;
+
+        assert(c);
+        assert(c->parameter);
+        assert(c->type == CONDITION_CONTROL_GROUP_CONTROLLER);
+
+        r = cg_mask_supported(&system_mask);
+        if (r < 0) {
+                log_error_errno(r, "Failed to determine supported controllers: %m");
+                return -EBADF;
```

Comment:
hmm, these functions are "api-like", i.e. are not supposed to log above LOG_DEBUG on their own, but propagate errors and let their callers do that. I see that condition_test_needs_update() is not doing that properly, but that's a mistake, and should be fixed. Anyway, for new code, please downgrade the message above log_error_errno() → log_debug_errno(). 

Also, please propagate the error you received, do not make up EBADF here... Note that log_debug_errno() returns the error you pass in hence you can write:

```
if (r < 0)
        return log_debug_errno(r, …);
```